leadbutt.py

```diff
@@ -85,6 +87,170 @@ def get_options(config_options, local_options, cli_options):
+            "writeIOsPS": "IO/Second"
+        },
+        "fileSys":
+        {
+            "used": "Kilobytes",
+            "usedFiles": "Count",
+            "usedFilePercent": "Percent",
+            "maxFiles": "Count",
+            "total": "Kilobytes",
+            "usedPercent": "Percent"
+        },
+        "processList":
+        {
+            "vss": "Kilobytes",
+            "tgid": "Count",
+            "parentID": "Count",
+            "memoryUsedPc": "Percent",
+            "cpuUsedPc": "Percent",
+            "id": "Count",
+            "rss": "Kilobytes"
+        }
+    }
+    return unit_map[category].get(statistic, 'Count')
+
+
+def output_log_results(formatter, ingestion_time, context, value):
+    metric_name = (formatter % context).replace('/', '.').lower()
+    line = '{0} {1} {2}\n'.format(
+        metric_name,
+        value,
+        ingestion_time,
+    )
+    sys.stdout.write(line)
+
+
+def process_log_results(results, options):
+    """
+    Output CW enhanced Monitoring to stdout.
+
+    http://boto.cloudhackers.com/en/latest/ref/logs.html
+    """
+
+    context = {}
+    # iterate over each result
+    for result in results:
+        # timestamp when metric arrived at queue
+        ingestion_time = result['ingestionTime'] / 1000
+        message = ast.literal_eval(result['message'])
+        context['Dimension'] = message['instanceID']
+        context['Namespace'] = 'AWS/RDS'
+        # iterate over category keys example: ["cpuUtilization", "memory"]
+        for category in message.keys():
+            context['MetricName'] = category
+            statistics = message[category]
+            statistics_type = type(statistics)
+            # process logs if value is a dictionary
+            if statistics_type is dict:
+                for statistic in statistics.keys():
+                    context['Statistic'] = statistic
+                    value = statistics[statistic]
+                    if type(value) is int or type(value) is float:
+                        # determine unit type example (Bytes/Second, Percent,
+                        # Count)
+                        context['Unit'] = unit_type_map(
+                            category, statistic)
+                        # output log to stdout for netcat to pickup
+                        output_log_results(options['Formatter'], ingestion_time, context, value)
+            # process list values differently, because of sub types
+            elif statistics_type is list:
+                for statistic_dict in statistics:
+                    statistic_list = statistic_dict.keys()
+                    # determine sub type using static map
+                    context['ListCategory'] = log_list_map(
+                        category, statistic_dict)
+                    for statistic in statistic_list:
+                        context['Statistic'] = statistic
+                        value = statistic_dict[statistic]
```

Comment:
There is a fair amount of code duplicated here, is there a way to refactor that?